# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/hardikjadli/flyrank/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


### My Lane

I chose the **Refresh / Content Opportunity Scoring** lane.

I want to identify which content pages should be reviewed first for refresh or improvement. This lane is a good fit because the starter dataset already contains observable search and content signals that can help rank pages for review. The goal is to support better content decisions rather than replacing human judgment.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


### Research Question

**Decision:** Which content pages should be reviewed first?

**Unit of analysis:** One content page.

**Output:** A ranked list of pages with a refresh opportunity score.

**Action:** The content or SEO team reviews the highest-ranked pages first for refresh, expansion, or monitoring.

**Cost of a wrong recommendation:** Time and effort may be spent reviewing the wrong pages while better opportunities are missed.

**Why ML can help:** The dataset contains many search and engagement signals. Machine learning can combine these signals to prioritize pages more effectively than simple manual rules.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [11]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Rows:", df.shape[0])
print("Columns:", df.shape[1])

print("\nTrend Direction Counts:")
print(df["trend_direction"].value_counts())

print("\nAverage 90-day Impressions:")
print(round(df["impressions_90d"].mean(), 2))

Rows: 30000
Columns: 44

Trend Direction Counts:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Average 90-day Impressions:
5200.37


In [10]:
!git clone https://github.com/hardikjadli/flyrank.git

%cd flyrank

!ls data/raw

Cloning into 'flyrank'...
remote: Enumerating objects: 140, done.
remote: Counting objects: 100% (140/140), done.
remote: Compressing objects: 100% (91/91), done.
remote: Total 140 (delta 52), reused 98 (delta 33), pack-reused 0 (from 0)
Receiving objects: 100% (140/140), 1.85 MiB | 8.11 MiB/s, done.
Resolving deltas: 100% (52/52), done.
/content/flyrank/flyrank
content_refresh_anonymized.csv


## Initial Observations

The dataset contains **30,000 rows** and **44 columns**. The average 90-day impressions are **5,200.37**, indicating that the dataset includes pages with varying levels of search traffic.

The `trend_direction` column shows that **16,262** pages are trending **down**, **5,962** are **stable**, **4,388** are trending **up**, **2,236** are **new**, and **1,152** are **flat**. Since a large number of pages are losing traffic, I want to investigate which features are associated with declining performance and whether they can help identify pages that should be refreshed first.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
### Careful Words



This work provides decision support based on observed data. It does not prove that refreshing a page will improve its performance or guarantee better search rankings. The recommendations are intended to help prioritize review and should always be checked by a human.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.